vulnfeeds/git/repository.go

```diff
@@ -61,6 +65,36 @@ type RepoTagsMap struct {
 // RepoTags acts as a cache for RepoTags results, keyed on the repo's URL.
 type RepoTagsCache map[string]RepoTagsMap
 
+// RemoteRepoRefsWithRetry will exponentially retry listing the peeled references of the repoURL up to retries times.
+func RemoteRepoRefsWithRetry(repoURL string, retries uint64) (refs []*plumbing.Reference, err error) {
+	remoteConfig := &config.RemoteConfig{
+		Name: "source",
+		URLs: []string{
+			repoURL,
+		},
+	}
+	repo := git.NewRemote(memory.NewStorage(), remoteConfig)
+
+	ctx := context.Background()
+
+	backoff := retry.NewExponential(1 * time.Second)
+	backoff = retry.WithMaxRetries(retries, backoff)
+
+	if err := retry.Do(ctx, backoff, func(ctx context.Context) error {
```

Comment:
nit: this seems like it could all be simplified as 

```go
err := retry.Do(
   ....)
return refs, err
```

without needing the extra nil checks and repetition.

Reply:
Hmm, I don't think so. There's an inline anonymous function that *only* returns an error, which is required by `retry.Do` and then there's the outer function that returns the refs and the error.

Reply:
It's possible I'm misreading this, but could we just write this as:

```golang
err = retry.Do(ctx, backoff, func(ctx context.Context) error {
  ... anonymous inline function
});

return refs, err
```

?